openslides/core/static/js/core/core.js

```diff
@@ -770,22 +982,65 @@ angular.module('OpenSlidesApp.core.projector', ['OpenSlidesApp.core'])
                     console.error("Error for slide " + element.name + ": " + element.error)
                 }
             });
+            $scope.scroll = -10 * Projector.get(1).scroll;
+            $scope.scale = 100 + 20 * Projector.get(1).scale;
         });
     });
 })
 
-.controller('SlideCustomSlideCtrl', function($scope, Customslide) {
-    // Attention! Each object that is used here has to be dealt on server side.
-    // Add it to the coresponding get_requirements method of the ProjectorElement
-    // class.
-    var id = $scope.element.id;
-    Customslide.find(id);
-    Customslide.bindOne(id, $scope, 'customslide');
-})
+.controller('SlideCustomSlideCtrl', [
+    '$scope',
+    'Customslide',
+    function($scope, Customslide) {
+        // Attention! Each object that is used here has to be dealt on server side.
+        // Add it to the coresponding get_requirements method of the ProjectorElement
+        // class.
+        var id = $scope.element.id;
+        Customslide.find(id);
+        Customslide.bindOne(id, $scope, 'customslide');
+    }
+])
 
-.controller('SlideClockCtrl', function($scope) {
-    // Attention! Each object that is used here has to be dealt on server side.
-    // Add it to the coresponding get_requirements method of the ProjectorElement
-    // class.
-    $scope.serverOffset = Date.parse(new Date().toUTCString()) - $scope.element.context.server_time;
-});
+.controller('SlideClockCtrl', [
+    '$scope',
+    function($scope) {
+        // Attention! Each object that is used here has to be dealt on server side.
+        // Add it to the coresponding get_requirements method of the ProjectorElement
+        // class.
+        $scope.servertime = ( Date.now() / 1000 - $scope.serverOffset ) * 1000;
+    }
+])
+
+.controller('SlideCountdownCtrl', [
+    '$scope',
+    '$interval',
+    function($scope, $interval) {
```

Comment:
Somewhere in this function you have to add an interval cancel for the case the the Ctrl goes down. See note box here: https://docs.angularjs.org/api/ng/service/$interval